tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/UserEvents/AuthenticationHttpContextExtensionsIntegration.cs

```diff
@@ -54,42 +56,46 @@ internal static object OnAsyncMethodEnd<TTarget>(object returnValue, Exception e
         {
             var claimsPrincipal = state.State as ClaimsPrincipal;
             // https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims
-            var claimsToTestInSafeMode = new[] { ClaimTypes.NameIdentifier, ClaimTypes.Name, "sub" };
-            if (claimsPrincipal?.Claims != null && Security.Instance is { TrackUserEvents: true } security)
+            var claimsToTest = new[] { ClaimTypes.NameIdentifier, ClaimTypes.Name, "sub", ClaimTypes.Email, ClaimTypes.Name, };
```

Comment:
A performance improvement could be defining ClaimsToTest as a static hashset:

        private static readonly HashSet<string> ClaimsToTest = new HashSet<string> 
        { 
            ClaimTypes.NameIdentifier, ClaimTypes.Name, "sub", ClaimTypes.Email 
        };

Reply:
Fixed in 266b7a62b.